### DISCUSSION TITLE
Question: issue importing adapters while using esm output

### DISCUSSION THREAD

**Discussion by giacomorebonato** [trpc/trpc#2033] [Q&A] [answered]
I am trying to build a Node apps with ESM support.
Hence I have set in my `tsconfig.json`

```
    "module": "NodeNext",
    "rootDir": "./src",
    "moduleResolution": "NodeNext",
```

And in my `package.json`

```
"type": "module"
```

The import of `import { fastifyTRPCPlugin } from '@trpc/server/adapters/fastify'` because of the following error:
`Cannot find module '@trpc/server/adapters/fastify' or its corresponding type declarations.`

How would you solve this issue?

**Comment by sachinraja** [comment, marked as answer]
This will be solved in the next major (v10).

**Reply by JekRock** [reply]
Before the v10 release, is there a way, by any chance, to make it work with ES modules?

**Reply by KATT** [reply]
Either:

- use the v10 alpha and be or okay with poorly documented breaking changes, or
- spend time migration `main` to support esm

**Comment by giacomorebonato** [comment]
Thanks!
